File: 11-container-with-most-water/container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int cap = 0;
        int i = 0, j = height.Length-1;
        while(j>i)
        {
            int s = j-i;
            cap = Math.Max(cap, s*Math.Min(height[i], height[j]));
            if(height[i]>height[j]){
                j--;
            }
            else{
                i++;
            }
        }
        return cap;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i,j=0,len(height)-1
        cap=0
        while j>i:
            m=j-i
            cap=max(m*min(height[i],height[j]), cap)
            if height[i]>height[j]:
                j-=1
            else:
                i+=1
        return cap